### PR TITLE
DT-1537 Fix WRITE scope and Proxy User Audit

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/aop/connectionproxy/OracleConnectionAspect.java [hunk 1]
```diff
@@ -98,7 +98,7 @@ private void setContext(final Connection conn) throws SQLException {
                 "nomis_context.set_context('AUDIT_MODULE_NAME','%s'); \n" +
                 "nomis_context.set_client_nomis_context('%s', '%s', '%s', '%s'); \n" +
                 "END;",
-                "ELITE2_API",
+                "PRISON_API",
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/aop/connectionproxy/OracleConnectionAspect.java:101]: rename the module to new API name

File: src/main/java/uk/gov/justice/hmpps/prison/aop/ProxyUserAspect.java [hunk 2]
```diff
@@ -33,16 +37,20 @@ public Object controllerCall(final ProceedingJoinPoint joinPoint) throws Throwab
         var proxyUser = authenticationFacade.getCurrentUsername();
         try {
             if (proxyUser != null) {
-                log.debug("Proxying User: {} for {}->{}", proxyUser,
+                final var nomisStaffUser = Boolean.toString(service.isStaff(proxyUser));
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/aop/ProxyUserAspect.java:40]: Find out if username is in staff list

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/AgencyResource.java [hunk 3]
```diff
@@ -100,8 +99,7 @@ public Agency getAgency(@PathVariable("agencyId") @ApiParam(value = "The ID of t
         @ApiResponse(code = 500, message = "Unrecoverable error occurred whilst processing request.", response = ErrorResponse.class)})
     @ApiOperation(value = "Update an existing agency", notes = "Requires MAINTAIN_REF_DATA")
     @PutMapping("/{agencyId}")
-    @HasWriteScope
-    @PreAuthorize("hasRole('MAINTAIN_REF_DATA')")
+    @PreAuthorize("hasRole('MAINTAIN_REF_DATA') and hasAuthority('SCOPE_write')")
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/AgencyResource.java:102]: Can't have two PreAuthorize on same method so have combined for now and remove HasWriteScope

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/ExtendedAuditableEntity.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.data.annotation.LastModifiedDate;
+
+import javax.persistence.Column;
+import javax.persistence.MappedSuperclass;
+import java.sql.Timestamp;
+
+@Getter(AccessLevel.PROTECTED)
+@Setter(AccessLevel.PROTECTED)
+@MappedSuperclass
+abstract class ExtendedAuditableEntity extends AuditableEntity {
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/ExtendedAuditableEntity.java:16]: Added for future work where we populate this data without the triggers

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderEvent.java [hunk 5]
```diff
@@ -43,30 +45,6 @@
     @Column(name = "EVENT_DATA_3", length = 4000)
     private String eventData3;
 
-    @Column(name = "CREATE_USER_ID")
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderEvent.java:46]: Don't repeat this

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/AgencyResource.java (hunk 3 above)
Comment (mikehalmamoj) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/AgencyResource.java:102]: Good find - that was my bad I think!

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**